integration/all_stores_test.go

```diff
@@ -0,0 +1,98 @@
+
+	storeConfigs := make([]storeConfig, len(stores))
+	for i, store := range stores {
+		storeConfigs[i] = storeConfig{From: oldestStoreStartTime.Add(time.Duration(i) * perStoreDuration).Format("2006-01-02"), IndexStore: store}
+	}
+
+	// bigtable client needs to set an environment variable when connecting to an emulator
+	bigtableFlag := map[string]string{"BIGTABLE_EMULATOR_HOST": bigtable.NetworkHTTPEndpoint()}
+
+	// here we are starting and stopping table manager for each index store
+	// this is a workaround to make table manager create tables for each config since it considers only latest schema config while creating tables
+	for i := range storeConfigs {
+		require.NoError(t, writeFileToSharedDir(s, cortexSchemaConfigFile, []byte(buildSchemaConfigWith(storeConfigs[i:i+1]))))
+
+		tableManager := e2ecortex.NewTableManager("table-manager", mergeFlags(ChunksStorageFlags, map[string]string{
+			"-table-manager.retention-period": "2520h", // setting retention high enough
+		}), "")
+		tableManager.HTTPService.SetEnvVars(bigtableFlag)
+		require.NoError(t, s.StartAndWaitReady(tableManager))
+
+		// Wait until the first table-manager sync has completed, so that we're
+		// sure the tables have been created.
+		require.NoError(t, tableManager.WaitSumMetrics(e2e.Greater(0), "cortex_dynamo_sync_tables_seconds"))
+		require.NoError(t, s.Stop(tableManager))
+	}
+
+	// Start rest of the Cortex components.
+	require.NoError(t, writeFileToSharedDir(s, cortexSchemaConfigFile, []byte(buildSchemaConfigWith(storeConfigs))))
+
+	ingester := e2ecortex.NewIngester("ingester", consul.NetworkHTTPEndpoint(), ChunksStorageFlags, "")
+	ingester.HTTPService.SetEnvVars(bigtableFlag)
+
+	distributor := e2ecortex.NewDistributor("distributor", consul.NetworkHTTPEndpoint(), ChunksStorageFlags, "")
+	querier := e2ecortex.NewQuerier("querier", consul.NetworkHTTPEndpoint(), ChunksStorageFlags, "")
+	querier.HTTPService.SetEnvVars(bigtableFlag)
+
+	require.NoError(t, s.StartAndWaitReady(distributor, ingester, querier))
+
+	// Wait until both the distributor and querier have updated the ring.
+	require.NoError(t, distributor.WaitSumMetrics(e2e.Equals(512), "cortex_ring_tokens_total"))
+	require.NoError(t, querier.WaitSumMetrics(e2e.Equals(512), "cortex_ring_tokens_total"))
+
+	// Push and Query some series to Cortex for each day starting from oldest start time from configs until now so that we test all the Index Stores
+	for ts := oldestStoreStartTime; ts.Before(now); ts = ts.Add(24 * time.Hour) {
+		series, expectedVector := generateSeries("series_1", ts)
+
+		c, err := e2ecortex.NewClient(distributor.HTTPEndpoint(), "", "", "user-1")
+		require.NoError(t, err)
+
+		res, err := c.Push(series)
+		require.NoError(t, err)
+		require.Equal(t, 200, res.StatusCode)
+
+		// Query the series both from the querier and query-frontend (to hit the read path).
+		c, err = e2ecortex.NewClient("", querier.HTTPEndpoint(), "", "user-1")
```

Comment:
This test should hit the storage, but how is it guaranteed that the series have been flushed to the storage and offloaded from the ingesters memory at this point?

I've the feeling we're not hitting the backend store neither on the write and read path, because the chunks are just kept in the ingesters memory.